### ISSUE TITLE
Failed to build PDF documentation for master branch

### ISSUE THREAD

**Issue by SchweinDeBurg** [mpv-player/mpv#1605]
CentOS 6.6 32-bit, Python version 2.6.6, rst2pdf version 0.93, ReportLab version 2.7, Waf version 1.8.5.

Traceback (most recent call last):
  File "/usr/bin/rst2pdf", line 9, in <module>
    load_entry_point('rst2pdf==0.93.dev', 'console_scripts', 'rst2pdf')()
  File "/usr/lib/python2.6/site-packages/rst2pdf/createpdf.py", line 1495, in main
    compressed=options.compressed)
  File "/usr/lib/python2.6/site-packages/rst2pdf/createpdf.py", line 653, in createPdf
    pdfdoc.multiBuild(elements)
  File "/usr/lib/python2.6/site-packages/reportlab/platypus/doctemplate.py", line 972, in multiBuild
    self.build(tempStory, **buildKwds)
  File "/usr/lib/python2.6/site-packages/reportlab/platypus/doctemplate.py", line 892, in build
    self.handle_flowable(flowables)
  File "/usr/lib/python2.6/site-packages/rst2pdf/createpdf.py", line 801, in handle_flowable
    self.handle_frameEnd()
  File "/usr/lib/python2.6/site-packages/reportlab/platypus/doctemplate.py", line 629, in handle_frameEnd
    self.handle_pageEnd()
  File "/usr/lib/python2.6/site-packages/reportlab/platypus/doctemplate.py", line 570, in handle_pageEnd
    raise LayoutError(ident)
reportlab.platypus.doctemplate.LayoutError: More than 10 pages generated without content - halting layout.  Likely that a flowable is too large for any frame.

**Comment by wm4** [comment]
Sporadic failure. rst2pdf can randomly fail if the layout is "inconvenient".

**Comment by SchweinDeBurg** [comment]
So your suggestion is try to rebuild?

**Comment by wm4** [comment]
No, that won't help. It's pretty much random, but it's sufficiently deterministic if the source files don't change. Last time we had this issue, moving some text around fixed it, and nobody knew why (see f6e05b14b26549c4beaaf754149c99fa20b11bd3).

**Comment by SchweinDeBurg** [comment]
Oh-h-h, it is really black magic... So I will wait for another magic move from you...

**Comment by mc4man** [comment]
Maybe it has to do with python or python-reportlab versions? Never have seen a build fail or pdf creation fail here. (several hundred builds
That would also  include 00b2611 when --log-file=<path> ect. was at the end of the Program Behavior section above Video
using python 2.7.6, python-reportlab 3.0

**Comment by lachs0r** [comment]
I can confirm that it works with newer ReportLab (3.1 here). So we should 
probably just make this a requirement.

**Comment by wm4** [comment]
> So we should probably just make this a requirement.

But how?

**Comment by lachs0r** [comment]
Custom check that runs Python 2 and does something like this:

```python
import sys, reportlab
sys.exit(int(reportlab.Version.split(".")[0]) < 3)
```

**Comment by SchweinDeBurg** [comment]
As I see Repotlab 3.x requires at least Python version 2.7, but under CentOS 6 there is Python 2.6.6. :-( BTW I never never have seen a PDF creation fail before while rebuilding master branch almost every day.

**Comment by lachs0r** [comment]
Frankly, CentOS is about the last distro I would use for multimedia stuff.
Probably the last distro I would use period. It really isn’t up to the task. 
Python 2.7 is now ALMOST 5 YEARS OLD. Support for Python 2.6 has ended in 
October 2013.

**Comment by SchweinDeBurg** [comment]
OK, I understand your point but on my work I am developing for CentOS 6 and prefer to use the same environment in production and personal laptop. So I will try to resolve this problem by myself.

**Comment by SchweinDeBurg** [comment]
I've solved my issue - so closing it. Here is solution: http://toomuchdata.com/2014/02/16/how-to-install-python-on-centos/ After this side-by-side installation to /opt/PSW/Python-2.7.9 I've installed needed packages using pip, then rebuilt "Waf" with Python 2.7 and manually copied its files to the Python 2.7 hierarchy. Then I've adjusted PATH for "Mpv" build time - and everything was succeeded.

BTW, CentOS is great for multimedia to - if you are software developer. :-D Just several repos added, just several MM packages (re)built by hands - and every media is available to view/listen.
